samples/snippets/core/testing/unit-testing-best-practices/csharp/unit-testing-best-practices/StringCalculator.cs

```diff
@@ -49,7 +49,7 @@ private static int HandleSingleNumber(string number)
             var result = int.TryParse(number, out var parsedNumber);
             if (result == false)
             {
-                 throw new ArgumentException(nameof(number));
+                 throw new ArgumentException(null, nameof(number));
```

Comment:
```suggestion
                 throw new ArgumentException(
                     $"Unable to parse '{numbers}' as an Int32.", nameof(number));
```

And something like this?

Reply:
My choice was to keep it functionally equivalent to how it was, but the null as first parameter does look weird and would raise some question marks. This is better.